h/api/schemas.py

```diff
@@ -59,6 +61,119 @@ class AnnotationSchema(JSONSchema):
+                                },
+                            },
+                        },
+                    },
+                    'highwire': {
+                        'type': 'object',
+                        'properties': {
+                            'doi': {
+                                'type': 'array',
+                                'items': {
+                                    'type': 'string',
+                                },
+                            },
+                        },
+                    },
+                    'link': {
+                        'type': 'array',
+                        'items': {
+                            'type': 'object',
+                            'properties': {
+                                'href': {
+                                    'type': 'string',
+                                },
+                                'type': {
+                                    'type': 'string',
+                                },
+                            },
+                            'required': [
+                                'href',
+                            ],
+                        },
+                    },
+                },
+            },
+            'group': {
+                'type': 'string',
+            },
+            'permissions': {
+                'title': 'Permissions',
+                'description': 'Annotation action access control list',
+                'type': 'object',
+                'patternProperties': {
+                    '^(admin|delete|read|update)$': {
+                        'type': 'array',
+                        'items': {
+                            'type': 'string',
+                            'pattern': '^(acct:|group:).+$',
+                        },
+                    }
+                },
+                'required': [
+                    'read',
+                ],
+            },
+            'references': {
+                'type': 'array',
+                'items': {
+                    'type': 'string',
+                },
+            },
+            'tags': {
+                'type': 'array',
+                'items': {
+                    'type': 'string',
+                },
+            },
+            'target': {
+                'type': 'array',
+                'items': [
+                    {
+                        'type': 'object',
+                        'properties': {
+                            'selector': {
+                            },
+                        },
+                        'required': [
+                            'selector',
+                        ],
+                    },
+                ],
+            },
+            'text': {
+                'type': 'string',
+            },
+            'uri': {
+                'type': 'string',
+            },
+        },
+        'required': [
+            'permissions',
```

Comment:
This one is actually okay because the `validate()` method does `new_appstruct['target_uri'] = appstruct.pop('uri', '')`.

In the future we probably should add uri to the required fields I agree, because why would we want people posting annotations with no uri? But for this pr, I just wanted to put in the absolute minimum validation to stop users from being able to crash it.